gobblin-compaction/src/main/java/gobblin/compaction/conditions/MultiRecompactionCondition.java

```diff
@@ -0,0 +1,34 @@
+package gobblin.compaction.conditions;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import gobblin.compaction.dataset.DatasetMetric;
+
+/**
+ * An implementation {@link RecompactionCondition} which contains other recompact conditions.
+ * As long as one of those sub conditions returns true, this condition will return true
```

Comment:
Why does the class assume `Condition A OR Condition B OR Condition C` ? This can be the default but we need to allow a condition operator `OR/AND` should be enough